File: The Core/4. Loop Tunnel/29. Addition Without Carrying/additionWithoutCarrying.cs
int additionWithoutCarrying(int param1, int param2) {
    string num1 = param1.ToString();
    string num2 = param2.ToString();
    int sum = 0;
    
    for(int i = 0; i < Math.Max(num1.Length, num2.Length); i++){
        int d1 = 0, d2 = 0;
        
        if(i < num1.Length){
            d1 = int.Parse(num1[(num1.Length - 1) - i].ToString());
        }
        
        if(i < num2.Length){
            d2 = int.Parse(num2[(num2.Length - 1) - i].ToString());
        }
        
        int product = d1+d2;
        
        if(product >= 10){
            product -=10;
        }
        
        sum = sum + product * Convert.ToInt32(Math.Pow(10, i));
    }
    
    return sum;
}

File: The Core/Loop Tunnel/29. Addition Without Carrying/additionWithoutCarrying.py
def additionWithoutCarrying(param1, param2):
    num1 = str(param1)
    num2 = str(param2)
    sum = 0
    
    for x in range(0, max(len(num1), len(num2))):
        d1 = 0
        d2 = 0
        
        if(x < len(num1)):
            d1 = str(int(num1[(len(num1) - 1) - x]))
            
        if(x < len(num2)):
            d2 = str(int(num2[(len(num2) - 1) - x]))

        product = int(d1) + int(d2)  
        if(product >= 10):
            product -= 10
        
        sum = sum + product * int(pow(10, x))
        
    return sum
